### PR TITLE
Remove uses of real domain names #371

### LINKED ISSUE #371: Remove uses of real domain names
In tests, such as `AddCommandTest.java`, test data include email addresses that are potentially genuine, e.g. `hans@google.com`.

This is problematic as [spam bots might target these emails](https://github.com/se-edu/addressbook-level4/pull/317#discussion_r108659852).

We should avoid the use of such email addresses. We could use the [example.com domain](https://en.wikipedia.org/wiki/Example.com) that is specifically created for examples instead.

### PR DESCRIPTION
**Pull request by chao1995**
Fixes #371 

Ready for review.

### CODE REVIEW

**Comment by CanIHasReview** [comment]
# v1

@chao1995 submitted v1 for review.

* [v1 1/1] [Replace potentially genuine email addresses with example\.com](https://github.com/se-edu/addressbook-level4/pull/375/commits/50dff56d1356c924c9fbe1ca133c463084426cbf)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/07367fab5054768f3243fb55eba1367d3d67a39b...50dff56d1356c924c9fbe1ca133c463084426cbf))
<details>
<summary>Checkout this PR version locally</summary>

```shell
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/375/1/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you want to fetch this PR to.

</details>

**Comment by chao1995** [comment]
@PierceAndy Looks like [validAddressBook.xml] has been covered? Include `EmailTest.java` in v2.

**Comment by CanIHasReview** [comment]
# v2

@chao1995 submitted v2 for review.

* [v2 1/1] [Replace potentially genuine email addresses with example\.com](https://github.com/se-edu/addressbook-level4/pull/375/commits/a4255a3b265f4fc2b9fd7cfb6e440aaca3bc09ba)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/07367fab5054768f3243fb55eba1367d3d67a39b...a4255a3b265f4fc2b9fd7cfb6e440aaca3bc09ba)) ([:chart_with_upwards_trend: Interdiff between v1 and v2](https://github.com/se-edu/addressbook-level4/compare/4bce2f8106341cfdadf63fe06467b2fb32c11e5a~1...4bce2f8106341cfdadf63fe06467b2fb32c11e5a))
<details>
<summary>Checkout this PR version locally</summary>

```shell
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/375/2/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you want to fetch this PR to.

</details>

**Comment by PierceAndy** [comment]
@chao1995 Yes you're right, validAddressBook.xml has been covered. My mistake, sorry.

File: src/main/java/seedu/address/model/util/SampleDataUtil.java [hunk 1]
```diff
@@ -18,22 +18,22 @@
...
                     getTagSet("colleagues", "friends")),
-                new Person(new Name("Charlotte Oliveiro"), new Phone("93210283"), new Email("charlotte@yahoo.com"),
+                new Person(new Name("Charlotte Oliveiro"), new Phone("93210283"), new Email("charlotte@example.com"),
                     new Address("Blk 11 Ang Mo Kio Street 74, #11-04"),
                     getTagSet("neighbours")),
-                new Person(new Name("David Li"), new Phone("91031282"), new Email("lidavid@google.com"),
+                new Person(new Name("David Li"), new Phone("91031282"), new Email("lidavid@example.com"),
                     new Address("Blk 436 Serangoon Gardens Street 26, #16-43"),
                     getTagSet("family")),
                 new Person(new Name("Irfan Ibrahim"), new Phone("92492021"), new Email("irfan@outlook.com"),
```
Comment (yamgent) [inline src/main/java/seedu/address/model/util/SampleDataUtil.java]: This email needs to be fixed too.

**Comment by yamgent** [comment]
Also you can propose a commit message.

**Comment by PierceAndy** [comment]
@yamgent Because there's only one commit, the merge commit message could use the commit's message.

**Comment by damithc** [comment]
@chao1995 this needs deconflicting

**Comment by CanIHasReview** [comment]
# v4

@chao1995 submitted v4 for review.

* [v4 1/1] [Replace potentially genuine email addresses with example\.com](https://github.com/se-edu/addressbook-level4/pull/375/commits/3a7b6e0cda53bc91d6333b23e95748f5e68180f9)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/8f4b6b4bb819c670f9039374e55781a848783676...3a7b6e0cda53bc91d6333b23e95748f5e68180f9)) ([:chart_with_upwards_trend: Interdiff between v3 and v4](https://github.com/se-edu/addressbook-level4/compare/db2822cbd73eb66bf450ba186175bdc91cee5e0d~1...db2822cbd73eb66bf450ba186175bdc91cee5e0d))
<details>
<summary>Checkout this PR version locally</summary>

```shell
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/375/4/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you want to fetch this PR to.

</details>

**Comment by chao1995** [comment]
@damithc Conflicting due to "Remove TestPerson" PR. fixed.

**Pull request merged**